internal/nodes/avi_model_l7_translator.go

```diff
@@ -655,3 +662,55 @@ func RemoveRedirectHTTPPolicyInModel(vsNode *AviVsNode, hostname, key string) {
 		}
 	}
 }
+
+func buildL7IngressInfraSetting(key string, vs *AviVsNode, vsvip *AviVSVIPNode, routeIgrObj RouteIngressModel) {
+	var found bool
+	var infraSetting *akov1alpha1.AviInfraSetting
+
+	var ingClassName string
+	if ingSpec, ok := routeIgrObj.GetSpec().(networkingv1beta1.IngressSpec); ok && ingSpec.IngressClassName != nil {
+		ingClassName = *ingSpec.IngressClassName
+	}
+
+	if !utils.GetIngressClassEnabled() {
+		return
+	} else if ingClassName == "" {
+		if ingClassName, found = isAviLBDefaultIngressClass(); !found {
+			return
+		}
+	}
+
+	if ingClassName != "" {
```

Comment:
I think we can either skip this check or return if ingClassName == "", because we don't have to call buildWithInfraSetting in that case